exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -175,52 +158,117 @@ func createElasticsearchBackoffFunc(config *RetrySettings) func(int) time.Durati
 	}
 }
 
-func shouldRetryEvent(status int) bool {
-	for _, retryable := range retryOnStatus {
-		if status == retryable {
-			return true
+func pushDocuments(ctx context.Context, index string, document []byte, bulkIndexer *esBulkIndexerCurrent) error {
+	return bulkIndexer.Add(ctx, index, bytes.NewReader(document))
+}
+
+func newBulkIndexer(logger *zap.Logger, client *elasticsearch7.Client, config *Config) (*esBulkIndexerCurrent, error) {
+	numWorkers := config.NumWorkers
+	if numWorkers == 0 {
+		numWorkers = runtime.NumCPU()
+	}
```

Comment:
[To reviewer] This is a change in default as well as the performance under default config.

NumCPU was the original default used by go-elasticsearch. But I think that setting the default to 1 makes more sense and 1 is also the default in opensearch exporter (see [code](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/opensearchexporter/log_bulk_indexer.go#L120)).

Reply:
Will change the default in a follow up PR